### PR TITLE
Boilerplate bootstrap 1 25bf1477435cc1d54ed1a0dff1c1f16405144db3

### PR DESCRIPTION
**Pull request by 2uasimojo**
### What type of PR is this?
refactor

### What this PR does / why we need it?
Standardization with boilerplate

### Which Jira/Github issue(s) this PR fixes?
[OSD-5645](https://issues.redhat.com/browse/OSD-5645)

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import "time"
 
 const (
 	OperatorName string = "managed-upgrade-operator"
+	OperatorNamespace string = "managed-upgrade-operator"
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply config/config.go:7]: For makefile purposes, boilerplate [derives the `OPERATOR_NAMESPACE` variable from this config.go](https://github.com/openshift/managed-upgrade-operator/pull/192/files/b5ce1d464dd2275ff2d99d342c3b5c95f1304e61#diff-8edb44c703eb01cf5c9b1ea79a0eb616eed990d8e2eb0a90113c7a913f948be6R3). This was previously [hardcoded in your project.mk](https://github.com/openshift/managed-upgrade-operator/pull/192/files/b5ce1d464dd2275ff2d99d342c3b5c95f1304e61#diff-0edc14c89325d031c18d96af1a592a5b7615d126c1ea7ba7333727ed4bfa3de9L3). So this change should be a no-op. (Honestly, I didn't even go look at whether/where that variable is actually consumed in the code or build process.)

**Comment by 2uasimojo** [comment]
Answering questions from [slack](https://coreos.slack.com/archives/CFJD1NZFT/p1611720038142200?thread_ts=1611593289.074100&cid=CFJD1NZFT) here:

> One note, I had a few issues with `make validate` - firstly the version of `yq` used seems rather old, I had to dig up an old version of that to make `validate` work - many of the arguments it tries to use are no longer compatible with recent versions of `yq`. Any reason it's dependent on an old version? (sorry if you've been asked this before)

It's using the [version we install in the boilerplate backing image](https://github.com/openshift/boilerplate/blob/884e5a831330ce731485fd84a70668453315b252/config/build.sh#L76-L86). This is one of the things boilerplate offers: controlled versions of tools like `yq` so operations like this are predictable.

In general, if you're having trouble with any `make` target -- particularly discrepancies with CI results -- boilerplate provides a utility that will run the target in a container environment as similar as possible to CI. In this example, you would run:

```shell
$ ./boilerplate/_lib/container-make validate
```

(I'm curious what version of `yq` you're using locally that you think is newer. AFAIU there are two different distributions of `yq` that have wildly different syntax, and don't really seem to be related to each other. FWIW the one in the boilerplate backing image is the same as the one behind `quay.io/app-sre/yq`, which is what muo (and everyone else) uses when [building the OLM catalog](https://github.com/openshift/managed-upgrade-operator/blob/937aef25c1ee09854c341ec0c31fe0967e885f41/hack/app_sre_create_image_catalog.sh#L31).)

> Secondly what version of operator-sdk did you use to regenerate the CRD in commit b5ce1d4? When I run `validate` (using SDK v0.17.0) it regenerates the CRD with a few formatting differences and subsequently `validate` complains because of that. This is not really a big issue, just curious.

Boilerplate [discovers and uses](https://github.com/openshift/boilerplate/blob/884e5a831330ce731485fd84a70668453315b252/boilerplate/openshift/golang-osd-operator/ensure.sh#L39-L42) the version of operator-sdk listed in your go.mod. Here's the relevant chunk of debug output when I run `validate` for muo:

```
++ go list -json -m github.com/operator-framework/operator-sdk
++ jq -r 'if .Replace != null then .Replace.Version else .Version end'
+ wantver=v0.17.0
+ echo 'go.mod says you want operator-sdk v0.17.0'
go.mod says you want operator-sdk v0.17.0
+ mkdir -p .operator-sdk/bin
+ cd .operator-sdk/bin
+ [[ -x ./operator-sdk ]]
+ which operator-sdk
which: no operator-sdk in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin:/go/bin)
+ case "$(uname -s)" in
++ uname -s
+ binary=operator-sdk-v0.17.0-x86_64-linux-gnu
+ which operator-sdk-v0.17.0-x86_64-linux-gnu
/usr/local/bin/operator-sdk-v0.17.0-x86_64-linux-gnu
+++ which operator-sdk-v0.17.0-x86_64-linux-gnu
++ realpath /usr/local/bin/operator-sdk-v0.17.0-x86_64-linux-gnu
+ osdk=/usr/local/bin/operator-sdk-v0.17.0-x86_64-linux-gnu
+ ln -sf /usr/local/bin/operator-sdk-v0.17.0-x86_64-linux-gnu operator-sdk
```

In this case, I suspect the formatting differences were because we do the [`yq` hack](https://github.com/openshift/boilerplate/pull/96) _after_ `operator-sdk generate`, and `yq` mungs some whitespace in addition to removing the offending field. IOW we are probably ending up with the same thing after `operator-sdk generate`, but `make validate` is doing (and validating) the additional `yq` step which is reformatting the CRD.

**Comment by mrbarge** [comment]
@2uasimojo Thanks for the feedback there, I appreciate the detailed reply to those questions!

Regarding `yq`, yes I was referring to the v4 branch which - as you highlight - has a [pretty different approach](https://mikefarah.gitbook.io/yq/v/v4.x/upgrading-from-v3) to just about everything. Looks like v3 is still supported until mid-2021, so no immediate issues there.

PR looks fine to me, I'll officially stamp it with an lgtm next week once we've first promoted some other MUO changes in the pipeline.

**Comment by mrbarge** [comment]
Sorry for the delay on me getting back to this @2uasimojo 
/lgtm

**Pull request merged**